service-worker.src.js

```diff
@@ -0,0 +1,17 @@
+const VULCANIZED_URLS = TO_BE_AUTOMATICALLY_REPLACED_WITH_URLS_ARRAY;
+const RESOLVED_VULCANIZED_URLS = VULCANIZED_URLS.map(
+  url => new URL(url, self.location).href
+);
+
+self.addEventListener('activate', event => {
+  // this allows to to intercept the very first request after SW installation
+  // without it, SW only takes effect after refreshing the page
+  clients.claim();
```

Comment:
I'm new to SW as well, but looking at the examples in MDN, it seems we may need to wait for the claim to finish https://developer.mozilla.org/en-US/docs/Web/API/Clients/claim#Example